supervisor/utils/dt.py

```diff
@@ -52,6 +32,9 @@ def parse_datetime(dt_str):
     Raises ValueError if the input is well formatted but not a valid datetime.
     Returns None if the input isn't well formatted.
     """
+    with suppress(ValueError, IndexError):
+        return ciso8601.parse_datetime(dt_str)
```

Comment:
Could we drop this helper class and use https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat instead ?

Reply:
the Core use that because it can not handle timezones 'Z'